subsys/net/ip/tcp.c

```diff
@@ -352,6 +364,27 @@ static void tcp_send(struct net_pkt *pkt)
 	tcp_pkt_unref(pkt);
 }
 
+#ifdef CONFIG_NET_TCP_RANDOMIZED_RTO
```

Comment:
I think we do not need to add the ifdef here

Reply:
Strictly not, but the compile settings don't like unused static functions, or am I missing something

Reply:
After the other changes, there will be no issues with the compiler. The unused functions will be optimized away by the compiler. Anyway, the idea with these kind of constructions is to allow the compiler to compile test even the unused functions which in turn can avoid bitrot even if the unused sections are left disabled for majority of the users.

Reply:
I applied your suggestion, does it make sense to run one of the socket/tcp test without randomized backoff to validate both paths?

Reply:
That would be great actually, validating both paths is a good idea.